text/0042-secret-debug-display.md

```diff
@@ -0,0 +1,88 @@
++ Feature name: `secret-debug-display`
++ Start date: 2020-07-03
++ RFC PR: [iotaledger/bee-rfcs#42](https://github.com/iotaledger/bee-rfcs/pull/42)
++ Bee issue: [iotaledger/bee#100](https://github.com/iotaledger/bee/issues/100)
+
+# Summary
+
+This RFC introduces two [procedural macros](https://doc.rust-lang.org/reference/procedural-macros.html), `SecretDebug`
+and `SecretDisplay`, to derive the traits [Debug](https://doc.rust-lang.org/std/fmt/trait.Debug.html) and
+[Display](https://doc.rust-lang.org/std/fmt/trait.Display.html) for secret material types in order to avoid leaking
+their internals.
+
+# Motivation
+
+Secret materials, private keys for example, are types that are expected to remain private and be shared/leaked under no
+circumstances. However, these types may be wrapped in other higher level types, a wallet for example, that may want to
+implement `Debug` and/or `Display`. Not implementing these traits on the secret types would then not permit it.
+At the same time, secret materials should not be exposed by logging them. The consensus is then to implement `Debug` and
+`Display` but to not actually log the inner secret. To make things easier, two procedural macros are proposed to
+automatically derive  such implementations.
+
+# Detailed design
+
+## Crate
+
+Since procedural macros require a crate defined as `proc-macro = true`, a new dedicated crate has to be created.
+
+## Derivation
+
+This feature is expected to be used by deriving `SecretDebug` and/or `SecretDisplay`.
```

Comment:
The only issue I can see is that deriving `SecretDisplay` implies completely giving up control over how the type is displayed. That might be a problem for some API users.